File: Models/EmployeeTimelineEntry.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;

namespace ExitSurveyAdmin.Models
***REMOVED***
    public class EmployeeTimelineEntry : BaseEntity
    ***REMOVED***
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int Id ***REMOVED*** get; set; ***REMOVED***

        [Required]
        public int EmployeeId ***REMOVED*** get; set; ***REMOVED***

        [JsonIgnore]
        public virtual Employee Employee ***REMOVED*** get; set; ***REMOVED***

        [Required]
        public string EmployeeActionCode ***REMOVED*** get; set; ***REMOVED***

        public virtual EmployeeActionEnum EmployeeAction ***REMOVED*** get; set; ***REMOVED***

        [Required]
        public string EmployeeStatusCode ***REMOVED*** get; set; ***REMOVED***
        public virtual EmployeeStatusEnum EmployeeStatus ***REMOVED*** get; set; ***REMOVED***

        [Required]
        public string Comment ***REMOVED*** get; set; ***REMOVED***
  ***REMOVED***
***REMOVED***

File: ClientApp/src/types/EmployeeTimelineEntry.ts
import ***REMOVED*** Transform ***REMOVED*** from 'class-transformer'
import ***REMOVED*** dateOrUndefined ***REMOVED*** from '../helpers/dateHelper'

export class EmployeeTimelineEntry ***REMOVED***
  id?: string
  employeeId?: string
  employeeActionCode?: string
  employeeStatusCode?: string
  comment?: string
  adminUserName?: string

  @Transform((***REMOVED*** value ***REMOVED***: ***REMOVED*** value: string ***REMOVED***) => dateOrUndefined(value, true))
  createdTs?: Date

  @Transform((***REMOVED*** value ***REMOVED***: ***REMOVED*** value: string ***REMOVED***) => dateOrUndefined(value, true))
  modifiedTs?: Date
***REMOVED***
